test/check-application

```diff
@@ -1292,6 +1294,96 @@ class TestFiles(testlib.MachineCase):
             b.click(".pf-v5-c-alert__action button")
             b.wait_not_present(".pf-v5-c-alert__action")
 
+    def testFileTypes(self):
+        m = self.machine
+        b = self.browser
+
+        exts = {
+            '': 'file',
+            '.tar.gz': 'archive-file',
+            '.ogg': 'audio-file',
+            '.py': 'code-file',
+            '.png': 'image-file',
+            '.txt': 'text-file',
+            '.mkv': 'video-file',
+        }
+
+        files = m.execute(fr"""
+            cd ~admin
+
+            # for each different file category, as per extension recognition...
+            for ext in {shlex.join(exts)}; do
+                # ... create one of each of the 7 fundamental types, minus symlinks
+                mkdir "dir$ext"
+                truncate -s1234 "file$ext"
+                mkfifo "fifo$ext"
+                python3 -c "import socket; socket.socket(socket.AF_UNIX).bind('sock$ext')"
+                mknod "chrdev$ext" c 0 0
+                mknod "blkdev$ext" b 0 0
+
+                # different types of broken symlink
+                ln -sf "loop$ext" "loop$ext"
+                ln -sf /bzzt "broken$ext"
+            done
+
+            # create some symlinks to those things
+            for source in *; do
+                ln -sf "$source" sym-"$source"
+                ln -sf "sym-$source" sym-sym-"$source"
+            done
+
+            ls /home/admin  # get the result
+        """).split()
+
+        # Make sure we created what we expected:
+        #   - len(exts) extension types; times
+        #   - 6 fundamental types plus 2 broken symlinks; times
+        #   - 3 levels of additional symlinking ('file', 'sym-file', 'sym-sym-file')
+        assert len(files) == len(exts) * (6 + 2) * 3
+
+        self.enter_files()
+        b.click("button[aria-label='Display as a list']")
+
+        # For each file we created, assert various things about how we expect
+        # it to be displayed.
+        for name in files:
+            selector = f"[data-item='{name}']"
+            classes = b.attr(selector, 'class').split()
+            assert b.text(f'{selector} .item-name') == name
```

Comment:
THis isn't pytest yet, so `self.assertEqual()` please for useful failures.

Reply:
done x4